plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ConnectionWrapper.java

```diff
@@ -269,7 +269,7 @@ public void refresh() {
 		updateWithResources(loadProjects());
 		state.set(LoadingState.LOADED);
 		fireChanged();
-		for (ProjectWrapper project : projects.values()) {
+		for (ProjectWrapper project : new ArrayList<>(projects.values())) {
```

Comment:
What's the rationale for this ?

Reply:
This is where a ConcurrentModificationException was triggered when you refresh a server twice in a row.
I included this small fix even though it's not related to the aim of this PR. I could separate it in terms of issue and PR if we agree on this being preferrable.